e2e/aks_model.go

```diff
@@ -201,7 +201,82 @@ func privateEndpointExists(ctx context.Context, t *testing.T, nodeResourceGroup,
 	return false, nil
 }
 
-func createPrivateEndpoint(ctx context.Context, t *testing.T, nodeResourceGroup, privateEndpointName string, vnet VNet) (armnetwork.PrivateEndpointsClientCreateOrUpdateResponse, error) {
+func createPrivateAzureContainerRegistry(ctx context.Context, t *testing.T, nodeResourceGroup, privateACRName string) error {
+	t.Logf("Creating private Azure Container Registry in rg %s\n", nodeResourceGroup)
+
+	acr, err := config.Azure.RegistriesClient.Get(ctx, nodeResourceGroup, privateACRName, nil)
+	if err == nil {
+		t.Logf("Private ACR already exists at id %s, skipping creation", *acr.ID)
+		return nil
+	}
+
+	// check if error is anything but not found
+	var azErr *azcore.ResponseError
+	if errors.As(err, &azErr) && azErr.StatusCode != 404 {
+		return fmt.Errorf("failed to get private ACR: %w", err)
+	}
+
+	t.Logf("ACR does not exist, creating...")
+	createParams := armcontainerregistry.Registry{
+		Location: to.Ptr(config.Config.Location),
+		SKU: &armcontainerregistry.SKU{
+			Name: to.Ptr(armcontainerregistry.SKUNamePremium),
+		},
+		Properties: &armcontainerregistry.RegistryProperties{
+			AdminUserEnabled:     to.Ptr(false),
+			AnonymousPullEnabled: to.Ptr(true), // required to pull images from the private ACR without authentication
+		},
+	}
+	pollerResp, err := config.Azure.RegistriesClient.BeginCreate(
```

Comment:
Only using BeginCreate not BeginUpdate since this ACR will only be created if it does not exist.